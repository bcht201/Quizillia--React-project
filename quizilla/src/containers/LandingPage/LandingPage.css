.LandingPageContainer{
    display: flex;
    flex-direction: column;
    height: 100%;
    align-items: center;
}

.quizButton {
    padding: 4px;
    border-radius: 10px;
    border: solid 2px yellowgreen;
}

.quizButton:hover{
    background-color: greenyellow;
    cursor: pointer;
}

.GameDropdown {
    margin-bottom: 5%;
    margin-top: 20%;
}




