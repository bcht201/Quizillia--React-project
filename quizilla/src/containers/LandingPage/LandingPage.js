import React from 'react';
import { Link} from 'react-router-dom';
import {Select, MenuItem} from '@material-ui/core';
import axios from 'axios';

class LandingPage extends React.Component{
    state = {
            categories:[],
            chosen_category: null    
    };

    componentDidMount(){
        axios.get("https://opentdb.com/api_category.php")
        .then(response => this.setState({categories: response.data.trivia_categories}));
    }

    handleChange = (event) => {
        this.setState({chosen_category: event.target.value});
    }

    onClickHandler =() => {
        this.state.chosen_category && this.props.pickCat(this.state.chosen_category);
    }

    render(){
        return (
            <div>
                <Link to="/quiz" className="quizButton" onClick={this.onClickHandler}>
                    <h2>Start Quiz!</h2>
                </Link>
                
                <Select
                    labelId="demo-simple-select-placeholder-label-label"
                    id="demo-simple-select-placeholder-label"
                    value= {this.state.category}
                    onChange= {(event) => this.handleChange(event)}
                    displayEmpty
                    >
                    {this.state.categories.map(category =>{
                        return(<MenuItem value={category.id}>{category.name}</MenuItem>)
                    })}
                </Select>
                <Select
                    labelId="demo-simple-select-placeholder-label-label"
                    id="demo-simple-select-placeholder-label"
                    value= {this.state.category}
                    onChange= {(event) => this.handleChange(event)}
                    displayEmpty
                    >
                    <MenuItem>Hard</MenuItem>
                    <MenuItem>Hard</MenuItem>
                    <MenuItem>Hard</MenuItem>
                    })}
                </Select>
            </div>
        )
    }

}

export default LandingPage;