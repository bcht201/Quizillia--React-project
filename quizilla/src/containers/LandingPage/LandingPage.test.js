import React from 'react';
import {shallow, mount} from 'enzyme';
import LandingPage from './LandingPage';
import { Link } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import {Select, MenuItem} from '@material-ui/core';

describe('LandingPage', () =>{
    let wrapper;
    const mockReset = jest.fn();
    beforeEach(()=>{
        wrapper = shallow(<LandingPage reset = {mockReset}/>)
    });

    it('should match snapshot', () =>{
        const mockReset = jest.fn();
        const tree = renderer.create(<Router><LandingPage reset = {mockReset}/></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('contains a 2 <div/> s, 1 is LandingPageContainer and the other is GameDropdown', () =>{
        expect(wrapper.find('div').length).toEqual(2);
        expect(wrapper.find('.LandingPageContainer').length).toEqual(1);
        expect(wrapper.find('.GameDropdown').length).toEqual(1);
    });

    it('should render our navbar', () => {
        expect(wrapper.find(Link).hasClass('quizButton')).toEqual(true);
    });

    it('should not redirect to /quiz when you click on start quiz link without selecting a category', () =>{
        const link = wrapper.find(Link);
        global.window = {location: {pathname : null}};
        expect(global.window.location.pathname).toEqual('/');
        link.simulate('click');
        expect(global.window.location.pathname).toEqual('/');
    });

    //STEPS:
    //create mock function 
    //shallow rendered LandingPage and gave it a prop called mockEventHandler
    //simulated a 'change' and on said change gave it mock values
    //now we write expect state to see if mock values are passed on into state after change
    // it('Should call pickCat (prop function) after setting state', () => {
    //     const mockEventHandler = jest.fn();
    //     const mockReset = jest.fn();
    //     const wrapper2 = mount(<Router>
    //         <LandingPage handleChange={mockEventHandler} reset = {mockReset} />
    //         </Router>)
    //     const event = {target: {
    //         name: 'chosen_category',
    //         value: 'val'
    //     }}
    //     // wrapper.setProps({onChange : mockEventHandler});
    //     // wrapper.find('#categoryDropdown').simulate('change', {target: {value: ["val"], name: ["chosen_category"]}})
    //     wrapper2.find('#categoryDropdown').simulate('change', event);

    //     // wrapper.update();
    //     // expect(wrapper.state().chosen_category).toEqual('val');
    //     expect(mockEventHandler.calledOnce).toBe(true);
    // })

    it('should change setstate after onchange', () =>{
        const fakeCatCall = jest.fn();
        const mockEvent = {
            target : {
                name: "chosen_category",
                value : "test"
            }
        };
        const expected = {
            categories:[],
            chosen_category: "test", 
            difficulty: null,
            question_count_data: null,
            max_questions: null,
            numberOfPlayers: 1
        };
        wrapper.instance().handleChange(mockEvent);
        expect(wrapper.state()).toEqual(expected);
    })

    // it('should display menu items after Api call', () => {
    //     loadData = async () => {

    //     }
    // })

    

    // it('should call pickCat propFunction after setting the state', () => {
    //     wrapper.update();
    //     wrapper.
    //     expect(wrapper.find('MenuItem').text()).toEqual('Pick a Category');
    // })

    it('should redirect to /quiz if a category is selected', () =>{
        // const wrapper2 = mount(<Router><LandingPage/></Router>);
        // wrapper2.setState({chosen_category: "entertainment: Whatever you want"});
        // const link = wrapper2.find(Link);
        // global.window = {location: {pathname : null}};
        // link.simulate('click');
        // expect(global.window.location.pathname).toEqual('/quiz');
    });

})