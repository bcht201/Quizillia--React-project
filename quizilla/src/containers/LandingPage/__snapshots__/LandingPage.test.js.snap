// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match snapshot 1`] = `
<div
  className="LandingPageContainer"
>
  <img
    alt="Logo"
    src="logo.png"
  />
  <div
    className="GameDropdown"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select-placeholder-label-label categoryDropdown"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="categoryDropdown"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Pick a category
      </div>
      <input
        name="chosen_category"
        onAnimationStart={[Function]}
        type="hidden"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select-placeholder-label-label difficultyDropdown"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="difficultyDropdown"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Pick a difficulty
      </div>
      <input
        name="difficulty"
        onAnimationStart={[Function]}
        type="hidden"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select-placeholder-label-label playerDropdown"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="playerDropdown"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Number of players
      </div>
      <input
        name="numberOfPlayers"
        onAnimationStart={[Function]}
        type="hidden"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <a
    className="quizButton"
    href="/quiz"
    onClick={[Function]}
  >
     
    <h2>
      Start Quiz!
    </h2>
     
  </a>
</div>
`;
